rotkehlchen/globaldb/upgrades/v12_v13.py

```diff
@@ -0,0 +1,93 @@
+            return
+
+        asset_updates, solana_tokens_data = [], []
+        duplicated_tokens = ('TRISIG', 'HODLSOL')  # TRISIG maps to TRISG & HODLSOL maps to HODL
+        with csv_file.open(encoding='utf-8') as f:
+            reader = csv.DictReader(f)
+            for row in reader:
+                if (old_id := row['old_id']) in duplicated_tokens:  # skip duplicate mapping
+                    continue
+
+                asset_updates.append(((new_id := f'solana/token:{row["address"]}'), old_id))
+                solana_tokens_data.append((
+                    new_id,
+                    'D',  # spl token
+                    row['address'],
+                    row['decimals'],
+                    None,
+                ))
+
+        write_cursor.switch_foreign_keys('OFF')
+        write_cursor.executemany('INSERT INTO solana_tokens(identifier, token_kind, address, decimals, protocol) VALUES(?, ?, ?, ?, ?)', solana_tokens_data)  # noqa: E501
+        write_cursor.executemany('UPDATE assets SET identifier = ? WHERE identifier = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE common_asset_details SET identifier = ? WHERE identifier = ?', asset_updates)  # noqa: E501
+
+        # duplicates were skipped during `solana_tokens` creation to avoid unique constraint errors
+        # but we still need to update any existing foreign key references to these old identifiers
+        asset_updates.extend([
+            ('solana/token:BLDiYcvm3CLcgZ7XUBPgz6idSAkNmWY6MBbm8Xpjpump', 'TRISIG'),
+            ('solana/token:58UC31xFjDJhv1NnBF73mtxcsxN92SWjhYRzbfmvDREJ', 'HODLSOL'),
+        ])
+        write_cursor.executemany('UPDATE asset_collections SET main_asset = ? WHERE main_asset = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE multiasset_mappings SET asset = ? WHERE asset = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE user_owned_assets SET asset_id = ? WHERE asset_id = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE price_history SET from_asset = ? WHERE from_asset = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE price_history SET to_asset = ? WHERE to_asset = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE binance_pairs SET base_asset = ? WHERE base_asset = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE binance_pairs SET quote_asset = ? WHERE quote_asset = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE counterparty_asset_mappings SET local_id = ? WHERE local_id = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE location_asset_mappings SET local_id = ? WHERE local_id = ?', asset_updates)  # noqa: E501
+        write_cursor.switch_foreign_keys('ON')
+
+        # delete the known duplicates.
+        write_cursor.execute('DELETE FROM assets WHERE identifier IN (?, ?)', duplicated_tokens)
```

Comment:
At the very end you should also add the index you created. Otherwise the index won't be there for all existing users